components/datadog/agentparams/filepermissions/permissions_nix.go

```diff
@@ -0,0 +1,82 @@
+package filepermissions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+	"github.com/DataDog/test-infra-definitions/common"
+)
+
+type UnixPermissionsOption = func(*UnixPermissions) error
+
+// UnixPermissions represents the owner, group, and permissions of a file.
+// Permissions are represented by a string (should be an octal). There is no check on the octal format.
+type UnixPermissions struct {
+	Owner       string
+	Group       string
+	Permissions string
+}
+
+var _ FilePermissions = (*UnixPermissions)(nil)
+
+// NewUnixPermissions creates a new UnixPermissions object and applies the given options.
+func NewUnixPermissions(options ...UnixPermissionsOption) optional.Option[FilePermissions] {
+	p, err := common.ApplyOption(&UnixPermissions{}, options)
+
+	if err != nil {
+		return optional.NewNoneOption[FilePermissions]()
+	}
+	return optional.NewOption[FilePermissions](p)
+}
+
+// SetupPermissionsCommand returns a command that sets the owner, group, and permissions of a file.
+func (p *UnixPermissions) SetupPermissionsCommand(path string) string {
+	var commands []string
+
+	if p.Owner != "" {
+		commands = append(commands, fmt.Sprintf("sudo chown %s %s", p.Owner, path))
+	}
+
+	if p.Group != "" {
+		commands = append(commands, fmt.Sprintf("sudo chgrp %s %s", p.Group, path))
+	}
+
+	if p.Permissions != "" {
+		commands = append(commands, fmt.Sprintf("sudo chmod %s %s", p.Permissions, path))
+	}
+
+	if len(commands) == 0 {
+		return ""
+	}
+	return fmt.Sprintf(`sudo sh -c "%v"`, strings.Join(commands, " && "))
+}
+
+// ResetPermissionsCommand returns a command that resets the owner, group, and permissions of a file to default.
+func (p *UnixPermissions) ResetPermissionsCommand(path string) string {
+	return fmt.Sprintf("sudo chown $USER %s && sudo chmod 644 %s", path, path)
```

Comment:
```suggestion
	return fmt.Sprintf("sudo chown $USER:$(id -gn) %s && sudo chmod 644 %s", path, path)
```
If you want to be super specific this will also reset the group 😄